src/Analyzer/Passes/IfTransformStringsToEnumPass.cpp

```diff
@@ -0,0 +1,306 @@
+#include <Analyzer/Passes/IfTransformStringsToEnumPass.h>
+
+#include <Analyzer/ConstantNode.h>
+#include <Analyzer/FunctionNode.h>
+#include <Analyzer/IQueryTreeNode.h>
+#include <Analyzer/InDepthQueryTreeVisitor.h>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/IDataType.h>
+
+#include <Functions/FunctionFactory.h>
+
+#include <stack>
+
+namespace DB
+{
+
+namespace
+{
+
+/// Visitor for finding functions that are used inside another function
+class FindUsedFunctionsVisitor : public ConstInDepthQueryTreeVisitor<FindUsedFunctionsVisitor>
+{
+public:
+    FindUsedFunctionsVisitor(
+        std::unordered_set<std::string_view> & used_functions_,
+        const std::unordered_set<std::string> & function_names_,
+        size_t stack_size_)
+        : used_functions(used_functions_), function_names(function_names_), stack_size(stack_size_)
+    {
+    }
+
+    static bool needChildVisit(VisitQueryTreeNodeType & parent, VisitQueryTreeNodeType & /* child */)
+    {
+        return parent->getNodeType() != QueryTreeNodeType::FUNCTION;
+    }
+
+    void visitImpl(const QueryTreeNodePtr & node)
+    {
+        auto * function_node = node->as<FunctionNode>();
+
+        if (!function_node)
+            return;
+
+        const auto & function_name = function_node->getFunctionName();
+        if (function_names.contains(function_name) && stack_size > 0)
+        {
+            const auto & alias = function_node->getAlias();
```

Comment:
After query analysis pass, all aliases from query expressions must be removed, except for table expressions. This is done because we want to simplify later steps. Not sure if this alias checks are necessary. Maybe we did not remove aliases somewhere, and it is bug in query analysis pass.